### PR TITLE
blast_repo fixes

### PR DESCRIPTION
**Pull request by devoncarew**
auto-publish

### CODE REVIEW

**Comment by devoncarew** [comment]
@lrhn - this PR updates dart-lang/characters to use some publishing automation. To use it, you'll want to merge this PR, then tag the merged commit w/ the above version tag (`v1.3.0`). You can go that from the command line and then push the tags to the repo, however, I generally create the tag by creating a new github release (which also creates a tag). That can be done from https://github.com/dart-lang/characters/releases.

**Review by kevmoo — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -131,3 +131,8 @@ Characters firstTagCharacters(Characters source) {
 [String]: https://api.dart.dev/stable/2.0.0/dart-core/String-class.html	"String class"
 [Uint16List]: https://api.dart.dev/stable/2.0.0/dart-typed_data/Uint16List-class.html	"Uint16List class"
 [Uint8List]: https://api.dart.dev/stable/2.0.0/dart-typed_data/Uint8List-class.html	"Uint8List class"
+
```
Comment (kevmoo) [inline README.md]: I'm meh on the update to the readme. It ONLY affects managers of the package, but it creates noise on every pub.dev page. I think we should drop it!

Comment (devoncarew) [reply README.md]: I can drop it. When contributing publishing automation to a mono-repo we can document on the main readme; it'll be helpful for people to better understand the workflow for contributing. And I'm less concerned now with adding repo smarts to all the individual single package repos, and instead make sure our mono-repos get all the latest automation.

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# A CI configuration to auto-publish pub packages.
+
+name: Publish
+
+on:
+  pull_request:
+    branches: [ master ]
+  push:
+    tags: [ 'v[0-9]+.[0-9]+.[0-9]+*' ]
```
Comment (lrhn) [inline .github/workflows/publish.yaml:9]: (I think this is a glorified glob pattern, not a regexp. Very confusing to read :thinking:.)

The pattern can drop the last `+`. It's probably benign, I don't think a glob pattern can get catastrophic backtracking, especially when there is nothing after the `*`. With any luck, the implementation will go to the `*` after seeing the first digit.

File: README.md (hunk 1 above)
Comment (lrhn) [reply README.md]: Consider having a `PUBLISHING.md` file with documentation about how to publish.
The `README.md` is for end-users, not developers.

File: .github/workflows/publish.yaml (hunk 2 above)
Comment (devoncarew) [reply .github/workflows/publish.yaml:9]: The pattern was intended to be able to capture `x.y.x-something`, for completeness. But thinking about it, if we really only ever intend to support publishing release versions via this automation, than dropping the suffix makes sense.

**Pull request merged**